From: Sebastian Keller <skeller@gnome.org>
Date: Mon, 21 Mar 2022 21:25:28 +0100
Subject: clutter: Mark clutter_stage_grab() return value as transfer full

The caller is expected to unref the returned grab and all C code already
does this, but grabs requested from js were resulting in the memory
getting leaked due to it being annotated as transfer none.

Fixes: https://gitlab.gnome.org/GNOME/mutter/-/issues/2189
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2345>

Origin: https://gitlab.gnome.org/GNOME/mutter/-/commit/703d8a77
---
 clutter/clutter/clutter-stage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clutter/clutter/clutter-stage.c b/clutter/clutter/clutter-stage.c
index 569df87..f34a424 100644
--- a/clutter/clutter/clutter-stage.c
+++ b/clutter/clutter/clutter-stage.c
@@ -3779,7 +3779,7 @@ G_DEFINE_BOXED_TYPE (ClutterGrab, clutter_grab,
  * Grabs input onto a certain actor. Events will be propagated as
  * usual inside its hierarchy.
  *
- * Returns: (transfer none): (nullable): an opaque #ClutterGrab handle, drop
+ * Returns: (transfer full): (nullable): an opaque #ClutterGrab handle, drop
  *   with clutter_grab_dismiss()
  **/
 ClutterGrab *
